### PR TITLE
(PUP-4698) Change fqdn_rand's return type to Integer

### PR DESCRIPTION
**Pull request by hlindberg**
Before this, fqdn_rand always returned a String. That is not the best
choice for a numeric value now that the evaluator is aware of different
data types as comparissons will typically be wrong (since '2' > '19'
when compared leicographically).

This fix adds a utility method that returns an integer version while the
old utility method producing a string is retained for those that use
Puppet as an API. The function now uses the integer version of this
utility method.

A test is added that asserts that an Integer is produced.

At some point it would be beneficial to reimplement the function using
the 4.x function API as it will simply raise a generic error if given
input of the wrong type. It will also magically change a floating point
input value to integer.

### CODE REVIEW

**Comment by hlindberg** [comment]
App Veyor failing with known issues regarding rdoc

File: spec/unit/parser/functions/fqdn_rand_spec.rb [hunk 1]
```diff
@@ -4,7 +4,11 @@
...
 describe "the fqdn_rand function" do
   include PuppetSpec::Scope
-  
+
+  it "returns an integer" do
+    expect(fqdn_rand(3)).to satisfy {|n| n.is_a?(Integer) }
+  end
+
   it "provides a random number strictly less than the given max" do
     expect(fqdn_rand(3)).to satisfy {|n| n.to_i < 3 }
```
Comment (kylog) [inline spec/unit/parser/functions/fqdn_rand_spec.rb]: can the `to_i` go away now?

**Comment by kylog** [comment]
Other than that one nit in the spec, I'm :+1:

**Comment by kylog** [comment]
And appveyor failures are the usual rdoc ones. Known bogo-failure.

File: spec/unit/parser/functions/fqdn_rand_spec.rb [hunk 2]
```diff
@@ -4,7 +4,11 @@
 
 describe "the fqdn_rand function" do
   include PuppetSpec::Scope
-  
+
+  it "returns an integer" do
+    expect(fqdn_rand(3)).to satisfy {|n| n.is_a?(Integer) }
```
Comment (thallgren) [inline spec/unit/parser/functions/fqdn_rand_spec.rb]: Simpler as `expect(fqdn_rand(3)).to be_a(Integer)`

**Comment by hlindberg** [comment]
ok - two nits - will update

**Pull request merged**